sdk/typescript/README.md

```diff
@@ -1,25 +1,83 @@
 
-If in an environment where `fetch` is not available, a polyfill must be used (see eg [node-fetch](https://www.npmjs.com/package/node-fetch)).
+- Status - For status and configuration details for the Gateway / Network.
+- Transaction - For transaction construction, preview, submission, and monitoring the status of an individual transaction.
+- Stream - For reading committed transactions.
+- State - For reading the current or past ledger state of the network.
+
+All APIs generated from OpenAPI specification are available after initialization either through `gatewayApi.lowLevel` or by calling `innerClient` on corresponding sub-api. See examples in [Low Level APIs section](#low-level-apis). On top of that, this package provides high-level counterparts for each sub api. High level APIs will grow over time as we start encountering repeating patterns when dealing with low level APIs. They are supposed to help with most commonly performed task and standardize ways of working with Gateway API and Typescript
+
+## Initialization
+
+Calling static `intialize` method from `GatewayApiClient` class will instantiate all low-level APIs together with their high-level equivalents and return them as a single object.
 
 ```typescript
-import "./node-fetch-polyfill" // Polyfill for fetch required if running in node-js
-import { StateApi, StatusApi, TransactionApi, StreamApi } from "@radixdlt/babylon-gateway-api-sdk";
+import { GatewayApiClient } from '@radixdlt/babylon-gateway-api-sdk'
+
+const gatewayApi = GatewayApiClient.initialize({
+  basePath: 'https://kisharnet-gateway.radixdlt.com',
+})
+```
+
+## Low Level APIs
 
-const transactionApi = new TransactionApi();
+Using generated APIs through `innerClient`
 
+```typescript
 async function getTransactionStatus(transactionIntentHashHex: string) {
-    let response = await transactionApi.transactionStatus({
-        transactionStatusRequest: {
-            intent_hash_hex: transactionIntentHashHex
-        }
-    });
-    return response.status;
+  let response = await gatewayApi.transaction.innerClient.transactionStatus({
+    transactionStatusRequest: {
+      intent_hash_hex: transactionIntentHashHex,
+    },
+  })
+  return response.status
+}
+```
+
+Using generated APIs through `lowLevel` object
+
+```typescript
+async function getEntityNonFungibleIds(
+  accountAddress: string,
+  vaultAddress: string,
+  nftAddress: string
+) {
+  return gatewayApi.lowLevel.state.entityNonFungibleIdsPage({
+    stateEntityNonFungibleIdsPageRequest: {
+      address: accountAddress,
+      vault_address: vaultAddress,
+      resource_address: nftAddress,
+    },
+  })
 }
 ```
+
+## State
+
+### Get Entity Details Vault Aggregated
+
+```typescript
+const entityDetails = await gatewayApi.state.getVaultEntityDetails(
+  'account_tdx_21_1p823h2sq7nsefkdharvvh5'
+)
+```
+
+## Transaction
+
+### Get Transaction Status
+
+```typescript
+const txStatus = await gatewayApi.transaction.getStatus(
```

Comment:
Nice :)

Could we show the user doing something with the status? EG console.log'ing the returned status message?